code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java

```diff
@@ -186,55 +176,74 @@ public void show() {
             fragmentFrameLayout.setId(frameLayoutResourceId);
         }
 
-        if (messagesMonitor != null) {
-            messagesMonitor.displayed();
-        }
-
         // replace the existing frame layout (if present) with a new MessageFragment
         final AEPMessage message = this;
 
         currentActivity.runOnUiThread(
-                new Runnable() {
-                    @Override
-                    public void run() {
-                        // add the frame layout to be replaced with the message fragment
-                        rootViewGroup.addView(fragmentFrameLayout);
-
-                        final FragmentManager fragmentManager =
-                                currentActivity.getFragmentManager();
-
-                        // ensure there are no existing webview fragments before creating a new one
-                        final Fragment currentMessageFragment =
-                                fragmentManager.findFragmentByTag(FRAGMENT_TAG);
-
-                        if (currentMessageFragment != null) {
-                            fragmentManager
-                                    .beginTransaction()
-                                    .remove(currentMessageFragment)
-                                    .commit();
-                        }
+                () -> {
+                    // add the frame layout to be replaced with the message fragment
+                    rootViewGroup.addView(fragmentFrameLayout);
```

Comment:
Can you check `messagesMonitor.show(message, delegateControl)` first and run all the UI logic later?